src/mapping_navigation/scripts/pure_pursuit.py

```diff
@@ -134,14 +123,16 @@ def search_target_index(self, state: State) -> Tuple[int, float]:
         Lf = self.look_forward_gain * state.velocity + self.look_ahead_distance
 
         # search look ahead target point index
-        while Lf > state.calc_distance(self.cx[ind], self.cy[ind]):
-            if (ind + 1) >= len(self.cx):
+        while Lf > state.calc_distance(self.path[ind].coordinate[0], self.path[ind].coordinate[1]):
+            if (ind + 1) >= len(self.path):
```

Comment:
Another one of those things that I should have asked earlier- why is this done? Did the previous while loop not find the next index to go towards?

Reply:
This is the adjustment for speed, its basically saying hey, if I am going fast is that last optimal points I found now too close to consider? If so look at the next point.